pubsub/cli.rb

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud"
+
+# Shows using gcloud to interact with Pub/Sub
+class PubSubSample
+  attr_accessor :topic, :sub
+
+  def initialize project
+    @project = project
+    @topic = "ruby-sample"
+    @sub = "ruby-sample-sub"
+    @subp = "ruby-sample-sub-p"
+
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+
+    topic = pubsub.topic @topic, autocreate: true
+
+    sub = topic.get_subscription @sub
+    topic.subscribe @sub if sub.nil?
+
+    subp = topic.get_subscription @subp
+    topic.subscribe(
+      @subp,
+      endpoint: "https://#{@project}.appspot.com/push") if subp.nil?
+  end
+
+  def publish message
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+    topic = pubsub.topic @topic
+    topic.publish message
+  end
+
+  def pull
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+    sub = pubsub.subscription @sub
+    messages = ""
+    sub.pull.each do |msg|
+      messages << msg.message.data
+      messages << "\n"
+      msg.acknowledge!
+    end
+    messages
+  end
+
+  def list_topics
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+    topics = pubsub.topics
+    topics_s = ""
+    topics.each do |topic|
+      topics_s << topic.name
+      topics_s << "\n"
+    end
+    topics_s
+  end
+
+  def list_subs
+    gcloud = Gcloud.new @projecta
+    pubsub = gcloud.pubsub
+    subs = pubsub.subscriptions
+    subs_s = ""
+    subs.each do |sub|
+      subs_s << sub.name
+      subs_s << "\n"
+    end
+    subs_s
```

Comment:
Instead of returning a string, I would expect `list_subscriptions` to print out subscription names.  It's simpler than concatenating a string and reads more clearly.

Also FYI if you wanted to get a newline delimited list of subscription names in Ruby, the idiomatic solution would be:

```ruby
def subscription_names
  gcloud = Gcloud.new @project
  pubsub = gcloud.pubsub

  pubsub.subscriptions.map(&:name).join "\n"
end
```